### PR TITLE
[5.1] Synchronizing the value env('APP_ENV') when the argument --env is passed to artisan.

### PR DESCRIPTION
**Pull request by loduis**
Synchronizing the value env('APP_ENV') when the argument --env is passed to artisan.

### CODE REVIEW

File: src/Illuminate/Foundation/EnvironmentDetector.php [hunk 1]
```diff
@@ -48,7 +48,10 @@ protected function detectConsoleEnvironment(Closure $callback, array $args)
         // and if it was that automatically overrides as the environment. Otherwise, we
         // will check the environment as a "web" request like a typical HTTP request.
         if (! is_null($value = $this->getEnvironmentArgument($args))) {
-            return head(array_slice(explode('=', $value), 1));
+            $env = head(array_slice(explode('=', $value), 1));
+            putenv("APP_ENV=$env");
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/EnvironmentDetector.php]: :-1:

Comment (taylorotwell) [inline src/Illuminate/Foundation/EnvironmentDetector.php]: @GrahamCampbell please explain your thumbs down.

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/EnvironmentDetector.php]: Feels wrong to call that function out of the blue. We never call that function in laravel at all. I just did a search. The environment detector is meant to detect the environment, not modify it.

Comment (loduis) [inline src/Illuminate/Foundation/EnvironmentDetector.php]: I was trying to run a command with the option --env
but I had a condition within the command as follows
```php
if (env('APP_ENV') == 'testing')
```
Clearly I had to change the line to:
```php
if (app()->enviroment ('local'))
```

```php
if ($this->laravel->environment ('local'))
```
Why env('APP_ENV') had the value **local**

**Comment by xujif** [comment]
@loduis  u can test laravel with docker. it's easy to change environment with commond line.

**Comment by GrahamCampbell** [comment]
> if (app()->enviroment('local'))

That's the correct way to do things, yes.

**Comment by GrahamCampbell** [comment]
The env variables should be read by the framework at a higher level, then not used by the user, in the same way we don't expect people to try to use the `$_GET` superglobal.

**Comment by loduis** [comment]
@GrahamCampbell

See this line: https://github.com/laravel/framework/blob/5.1/src/Illuminate/Foundation/Bootstrap/DetectEnvironment.php#L26

```php

        $app->detectEnvironment(function () {
            return env('APP_ENV', 'production');
        });
```

This in sync the value of **env('APP_ENV')** helper and so has the framework
It would have been better return only **production**

Agree with you, it is better to use code
```php
if (app()->enviroment('local'))
```

And probably it would have been better to adjust the code like this:

```php
return env('APP_ENV', head(array_slice(explode('=', $value), 1)));
```
All method.

```php
    protected function detectConsoleEnvironment(Closure $callback, array $args)
    {
        // First we will check if an environment argument was passed via console arguments
        // and if it was that automatically overrides as the environment. Otherwise, we
        // will check the environment as a "web" request like a typical HTTP request.
        if (! is_null($value = $this->getEnvironmentArgument($args))) {
            return env('APP_ENV', head(array_slice(explode('=', $value), 1)));
        }

        return $this->detectWebEnvironment($callback);
    }
```

**Pull request closed without merging**